compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -931,7 +931,29 @@ OMR::Power::TreeEvaluator::mLastTrueEvaluator(TR::Node *node, TR::CodeGenerator
 TR::Register*
 OMR::Power::TreeEvaluator::mToLongBitsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
+   TR_ASSERT_FATAL_WITH_NODE(node, node->getDataType().getVectorLength() == TR::VectorLength128,
+                             "Only 128-bit vectors are supported %s", node->getDataType().toString());
+
+   if (node->getDataType().getVectorElementType() != TR::Int8)
+     {
+     TR_ASSERT_FATAL(false, "Unsupported vector type %s for mToLongBits\n", node->getDataType().toString());
+     }
+
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *srcReg = cg->evaluate(firstChild);
+   TR::Register *tmpReg = cg->allocateRegister(TR_VRF);
+   TR::Register *resReg = cg->allocateRegister();
+
+   if (cg->comp()->target().cpu.isLittleEndian())
+      generateTrg1Src1Instruction(cg, OMR::InstOpCode::xxbrq, node, tmpReg, srcReg);
+
+   generateTrg1Src1Instruction(cg, OMR::InstOpCode::vextractbm, node, resReg, tmpReg);
```

Comment:
logically incorrect, since temReg is undefined for BigEndian (plus, tmpReg is not needed for BigEndian to begin with).

higher-level question about Vector API:  is it defined to be BigEndian data?  for this particular case, the mask is expected to reflect BigEndian mask.  i.e.  there is a performance penalty to run on LittleEndian always.